scheduler/plugin/plugin.go

```diff
@@ -41,12 +42,28 @@ func New(configuration runtime.Object, frameworkHandle framework.Handle) (framew
 		return nil, fmt.Errorf("failed to decode into %s PluginConfig: %w", PluginName, err)
 	}
 
-	return NewFromConfig(context.Background(), config)
+	plugin, err := NewFromConfig(context.Background(), config)
+	return maskInterfaces(plugin), err
+}
+
+// maskInterfaces ensures the caller can do type checking to detect what the
+// plugin supports.
+func maskInterfaces(plugin *wasmPlugin) framework.Plugin {
+	if plugin == nil {
+		return nil
+	}
+	if plugin.guestExports == exportFilterPlugin {
```

Comment:
This is how we support current day plugin use, where a type assertion happens on the result of a pluginFactory.

as we support more plugin types, this logic will get more complicated, a cartesian product perhaps of many combinations. However, the cardinality of plugin types is fixed.